rust/operator-binary/src/druid_controller.rs

```diff
@@ -1156,24 +1147,37 @@ fn build_rolegroup_statefulset(
         );
 
     if merged_rolegroup_config.logging.enable_vector_agent {
-        pb.add_container(
-            product_logging::framework::vector_container(
-                resolved_product_image,
-                DRUID_CONFIG_VOLUME_NAME,
-                LOG_VOLUME_NAME,
-                merged_rolegroup_config
-                    .logging
-                    .containers
-                    .get(&Container::Vector),
-                ResourceRequirementsBuilder::new()
-                    .with_cpu_request("250m")
-                    .with_cpu_limit("500m")
-                    .with_memory_request("128Mi")
-                    .with_memory_limit("128Mi")
-                    .build(),
-            )
-            .context(BuildVectorContainerSnafu)?,
-        );
+        match druid
+            .spec
+            .cluster_config
+            .vector_aggregator_config_map_name
+            .to_owned()
+        {
+            Some(vector_aggregator_config_map_name) => {
+                pb.add_container(
+                    product_logging::framework::vector_container(
+                        resolved_product_image,
+                        DRUID_CONFIG_VOLUME_NAME,
+                        LOG_VOLUME_NAME,
+                        merged_rolegroup_config
+                            .logging
+                            .containers
+                            .get(&Container::Vector),
+                        ResourceRequirementsBuilder::new()
+                            .with_cpu_request("250m")
+                            .with_cpu_limit("500m")
+                            .with_memory_request("128Mi")
+                            .with_memory_limit("128Mi")
+                            .build(),
+                        &vector_aggregator_config_map_name,
```

Comment:
With the above change, this can be:

```suggestion
                        vector_aggregator_config_map_name,
```